hub-detect/src/main/resources/hub-detect-4-sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+# To override the default version key, specify your
+# own DETECT_VERSION_KEY in your environment and
+# *that* key will be used to get the download url from
+# artifactory.
+DETECT_VERSION_KEY=${DETECT_VERSION_KEY:-DETECT_LATEST_4}
+
+# You can specify your own download url from
+# artifactory which can bypass using the property keys
+# (this is mainly for QA purposes only)
+DETECT_SOURCE=${DETECT_SOURCE:-}
+
+# To override the default location of /tmp, specify
+# your own DETECT_JAR_PATH in your environment and
+# *that* location will be used.
+DETECT_JAR_PATH=${DETECT_JAR_PATH:-/tmp}
+
+# If you want to pass any java options to the
+# invocation, specify DETECT_JAVA_OPTS in your
+# environment. For example, to specify a 6 gigabyte
+# heap size, you would set DETECT_JAVA_OPTS=-Xmx6G.
+DETECT_JAVA_OPTS=${DETECT_JAVA_OPTS:-}
+
+# If you want to pass any additional options to
+# curl, specify DETECT_CURL_OPTS in your environment.
+# For example, to specify a proxy, you would set
+# DETECT_CURL_OPTS=--proxy http://myproxy:3128
+DETECT_CURL_OPTS=${DETECT_CURL_OPTS:-}
+
+SCRIPT_ARGS="$@"
+LOGGABLE_SCRIPT_ARGS=""
+
+for i in $*; do
+  if [[ $i == --blackduck.hub.password=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.hub.password=<redacted>"
+  elif [[ $i == --blackduck.hub.proxy.password=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.hub.proxy.password=<redacted>"
+  elif [[ $i == --blackduck.hub.api.token=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.hub.api.token=<redacted>"
+  elif [[ $i == --blackduck.password=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.password=<redacted>"
+  elif [[ $i == --blackduck.proxy.password=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.proxy.password=<redacted>"
+  elif [[ $i == --blackduck.api.token=* ]]; then
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS --blackduck.api.token=<redacted>"
+  else
+    LOGGABLE_SCRIPT_ARGS="$LOGGABLE_SCRIPT_ARGS $i"
+  fi
+done
+
+run() {
+  get_detect
+  run_detect
+}
+
+get_detect() {
+  if [ -z "${DETECT_SOURCE}" ]; then
+    if [ -z "${DETECT_RELEASE_VERSION}" ]; then
+      VERSION_CURL_CMD="curl ${DETECT_CURL_OPTS} --silent --header \"X-Result-Detail: info\" 'https://test-repo.blackducksoftware.com/artifactory/api/storage/bds-integrations-release/com/blackducksoftware/integration/hub-detect?properties=${DETECT_VERSION_KEY}' | grep \"${DETECT_VERSION_KEY}\" | sed 's/[^[]*[^\"]*\"\([^\"]*\).*/\1/'"
+      DETECT_SOURCE=$(eval $VERSION_CURL_CMD)
```

Comment:
If a bad option is set for DETECT_VERSION_KEY, this causes problems. Unsure the proper way to address that. Validate DETECT_VERSION_KEY? Validate setting DETECT_SOURCE?

DETECT_VERSION_KEY TEST:

$ export DETECT_VERSION_KEY=BOB_THE_BUILDER
$ ./dana-detect.sh 
will look for : 
You don't have the current file, so it will be downloaded.
getting  from remote
saved  to /tmp/
running detect: java  -jar /tmp/ 

$ unset DETECT_VERSION_KEY
$ ./dana-detect.sh 
will look for : https://test-repo.blackducksoftware.com/artifactory/bds-integrations-release/com/blackducksoftware/integration/hub-detect/4.3.0/hub-detect-4.3.0.jar
You don't have the current file, so it will be downloaded.
getting https://test-repo.blackducksoftware.com/artifactory/bds-integrations-release/com/blackducksoftware/integration/hub-detect/4.3.0/hub-detect-4.3.0.jar from remote
saved https://test-repo.blackducksoftware.com/artifactory/bds-integrations-release/com/blackducksoftware/integration/hub-detect/4.3.0/hub-detect-4.3.0.jar to /tmp/hub-detect-4.3.0.jar
running detect: java  -jar /tmp/hub-detect-4.3.0.jar

Reply:
We'll add the following:
  if [ -z "${DETECT_SOURCE}" ]; then
    echo "DETECT_SOURCE was not set and/or computed correctly, please check your configuration."
    exit -1
  fi